### PR TITLE
List subgroup members

### LINKED ISSUE #6: List members of subgroup
As a member in a group chat
I want the ability to list all the members within a subgroup (mini-groups)
So that I can know who I will alert

```
# Command
/list [subgroup]

# Response
[members of subgroup]
```

### PR DESCRIPTION
**Pull request by AshciR**
closes #6

### CODE REVIEW

File: src/shout_subgroup/list_subgroup.py [hunk 1]
```diff
@@ -6,12 +6,44 @@
...
-from shout_subgroup.repository import find_all_subgroups_in_group_chat
+from shout_subgroup.exceptions import NotGroupChatError, SubGroupDoesNotExistsError
+from shout_subgroup.models import SubgroupModel, UserModel
+from shout_subgroup.repository import (find_all_subgroups_in_group_chat,
+                                       find_subgroup_by_telegram_group_chat_id_and_subgroup_name,
+                                       find_all_users_in_subgroup)
 from shout_subgroup.utils import is_group_chat
 
 
+async def _handle_list_subgroups(update: Update, db: Session, telegram_group_chat_id: int):
```
Comment (AshciR) [inline src/shout_subgroup/list_subgroup.py:17]: Refactor: Moved this logic into it's own function b/c it makes the handler function succinct.

File: src/shout_subgroup/list_subgroup.py [hunk 2]
```diff
@@ -24,6 +56,44 @@ async def list_subgroups(db: Session, telegram_group_chat_id: int) -> list[Type[
     return subgroups
 
 
+async def _handle_list_subgroup_members(update: Update, db: Session, telegram_group_chat_id: int, subgroup_name: str):
```
Comment (AshciR) [inline src/shout_subgroup/list_subgroup.py:59]: Focus: New helper method for listing subgroup members. Responsible for sending the reply messages

File: src/shout_subgroup/list_subgroup.py [hunk 3]
```diff
@@ -24,6 +56,44 @@ async def list_subgroups(db: Session, telegram_group_chat_id: int) -> list[Type[
...
+
+    # "@username1, @username2, @username3"
+    usernames = [f"@{member.username}" for member in members]
+    joined_usernames = ", ".join(usernames)
+
+    await update.message.reply_text(f"'{subgroup_name}' subgroup has these members: {joined_usernames}")
+    return
+
+
+async def list_subgroup_members(db: Session, telegram_group_chat_id: int, subgroup_name: str) -> list[Type[UserModel]]:
```
Comment (AshciR) [inline src/shout_subgroup/list_subgroup.py:74]: Focus: Handles the service logic for listing the members

File: src/shout_subgroup/list_subgroup.py [hunk 4]
```diff
@@ -35,20 +105,24 @@ async def list_subgroup_handler(update: Update, context: ContextTypes.DEFAULT_TY
...
-        # "mock-subgroup-1, mock-subgroup-2, mock-subgroup-3"
-        subgroups_names = [sub.name for sub in subgroups]
-        joined_subgroup_names = ", ".join(subgroups_names)
+    args = context.args
+    chat_id = update.effective_chat.id
+    subgroup_name = args[0] if len(args) == 1 else ""
 
-        await update.message.reply_text(f"Here are the subgroups for this chat: {joined_subgroup_names}")
-        return
+    try:
```
Comment (AshciR) [inline src/shout_subgroup/list_subgroup.py:112]: Refactor: Using the helper methods mentioned above

File: tests/test_list_subgroup.py [hunk 5]
```diff
@@ -59,3 +59,81 @@ async def test_list_subgroup_throws_not_group_chat_exception(db: Session):
         await list_subgroups(db, user_chat_id)
 
     assert str(user_chat_id) in ex.value.message
+
+
+@pytest.mark.asyncio
+async def test_list_subgroup_members(db: Session):
```
Comment (AshciR) [inline tests/test_list_subgroup.py:65]: Focus: Added 4 tests for `/list <subgroup>`

**Review by rick1e — APPROVED** [review]

**Pull request merged**